comp110-worksheet-7/DirectoryUtils.cs

```diff
@@ -5,56 +5,109 @@
 using System.Text;
 using System.Threading.Tasks;
 
-namespace comp110_worksheet_7
-{
-	public static class DirectoryUtils
-	{
+namespace comp110_worksheet_7{
+
+	public static class DirectoryUtils{
 		// Return the size, in bytes, of the given file
-		public static long GetFileSize(string filePath)
-		{
+		public static long GetFileSize(string filePath){
 			return new FileInfo(filePath).Length;
 		}
 
 		// Return true if the given path points to a directory, false if it points to a file
-		public static bool IsDirectory(string path)
-		{
+		public static bool IsDirectory(string path){
 			return File.GetAttributes(path).HasFlag(FileAttributes.Directory);
 		}
 
 		// Return the total size, in bytes, of all the files below the given directory
-		public static long GetTotalSize(string directory)
-		{
-			throw new NotImplementedException();
+		public static long GetTotalSize(string directory){
+            long totalSize = 0;
+            // Getting all elements in the directory
+            string[] dicElements = Directory.GetFiles(directory, "*", SearchOption.AllDirectories);
+            //for each item in the array, it iterates 
+            foreach(string file in dicElements){
+                totalSize += GetFileSize(file);
+            }
+            // Returns the size of array
+            return totalSize;
 		}
 
 		// Return the number of files (not counting directories) below the given directory
-		public static int CountFiles(string directory)
-		{
-			throw new NotImplementedException();
+		public static int CountFiles(string directory){
+            string[] dicElements = Directory.GetFiles(directory, "*", SearchOption.AllDirectories);
+
+            return dicElements.Length;
 		}
 
-		// Return the nesting depth of the given directory. A directory containing only files (no subdirectories) has a depth of 0.
-		public static int GetDepth(string directory)
-		{
-			throw new NotImplementedException();
+		public static int GetDepth(int curDepth = 0, string directory){
```

Comment:
This is a compile error ("Optional parameters must appear after all required parameters") -- you would need to put `curDepth` after `directory` for this to work